src/redismodule.h

```diff
@@ -912,6 +916,7 @@ typedef int (*RedisModuleConfigSetNumericFunc)(const char *name, long long val,
 typedef int (*RedisModuleConfigSetBoolFunc)(const char *name, int val, void *privdata, RedisModuleString **err);
 typedef int (*RedisModuleConfigSetEnumFunc)(const char *name, int val, void *privdata, RedisModuleString **err);
 typedef int (*RedisModuleConfigApplyFunc)(RedisModuleCtx *ctx, void *privdata, RedisModuleString **err);
+typedef int (*RedisModuleCustomAuthCallback)(RedisModuleCtx *ctx, RedisModuleString *username, RedisModuleString *password, RedisModuleString **err);
```

Comment:
```suggestion
typedef int (*RedisModuleAuthCallback)(RedisModuleCtx *ctx, RedisModuleString *username, RedisModuleString *password, RedisModuleString **err);
```

Still going to suggest this naming. @oranagra Do you have any thoughts on this, adding "custom" to modules seems redundant.

Reply:
i'm not sure i'm an authority on that subject.
to me it feels ok, it's not that the module is custom, here, it's that it's a "custom auth", i.e. the module is intervening in another mechanism.
usually modules register commands, and data types, and configs, all of which only affect the module mechanisms (and the module's commands).

so maybe when the module affects redis, auth, file system, compression, network, we can call these custom?

i also don't have a problem with your suggestion to drop the "custom" part, i don't think it'll cause any confusion if we remove it.

Reply:
I made the renaming change here: https://github.com/redis/redis/pull/11659/commits/4d576c51b7f531b2a4367590e137133e30589022